ui/src/app/kvm/views/KVMDetails/LxdProject/ProjectVMs/VMsTable/StatusColumn/StatusColumn.tsx

```diff
@@ -0,0 +1,53 @@
+import { Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
+import DoubleRow from "app/base/components/DoubleRow";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine } from "app/store/machine/types";
+import { useFormattedOS } from "app/store/machine/utils";
+import type { RootState } from "app/store/root/types";
+import { NodeStatusCode } from "app/store/types/node";
+import { getPowerIcon } from "app/utils";
+
+type Props = {
+  systemId: Machine["system_id"];
+};
+
+const transientStatuses = [
+  NodeStatusCode.COMMISSIONING,
+  NodeStatusCode.DEPLOYING,
+  NodeStatusCode.DISK_ERASING,
+  NodeStatusCode.ENTERING_RESCUE_MODE,
+  NodeStatusCode.EXITING_RESCUE_MODE,
+  NodeStatusCode.RELEASING,
+  NodeStatusCode.TESTING,
+];
+
+const getIcon = (machine: Machine) => {
```

Comment:
I wonder if this and the `getPowerIcon` utility should be merged into a component that generates the icon/classes etc. as this is pretty much the same thing as is done in PowerColumn.

Reply:
Good idea! I'll add a card for it